BaragonService/src/main/java/com/hubspot/baragon/service/resources/AgentCheckinResource.java

```diff
@@ -47,7 +47,7 @@ public Response addAgent(@PathParam("clusterName") String clusterName, BaragonAg
       }
     } catch (Exception e) {
       LOG.error("Could not register agent startup", e);
-      return Response.status(Status.INTERNAL_SERVER_ERROR).entity(e.getMessage()).build();
+      return Response.status(Status.INTERNAL_SERVER_ERROR).entity(e).build();
```

Comment:
I would be concerned that not all exceptions serialize to JSON nicely, but not 100% sure